File: rules/javascript/tests/js.command-injection/aliased.js
// Aliased/namespaced imports that the name-based text query cannot match —
// only identity resolution catches these.
import { exec as run } from 'child_process';
import { spawn as launch } from 'child_process';

function viaAlias(req) {
  run("ls " + req.query.dir); // crit:expect js.command-injection
}

function viaAliasSpawn(req) {
  launch("ping " + req.query.host); // crit:expect js.command-injection
}

function unrelatedName(req) {
  const run2 = (x) => x;
  run2(req.query.x); // crit:expect-not js.command-injection
}

function sanitizedAlias(req) {
  const n = parseInt(req.query.n);
  run("head -n " + n); // crit:expect-not js.command-injection
}

File: rules/go/tests/go.cmd-injection/aliased.go
package main

import (
	"net/http"
	ex "os/exec"
)

// Aliased package import: the text query keyed on "exec.Command" cannot match
// `ex.Command`; identity resolution does.
func viaAlias(r *http.Request) {
	ex.Command("sh", "-c", "ping "+r.FormValue("host")) // crit:expect go.cmd-injection
}

func viaAliasContext(r *http.Request, ctx interface{}) {
	ex.CommandContext(ctx, "sh", "-c", r.URL.Query().Get("q")) // crit:expect go.cmd-injection
}

func fixedAlias() {
	ex.Command("ls", "-la") // crit:expect-not go.cmd-injection
}

func internalAlias(host string) {
	ex.Command("ping", host) // crit:expect-not go.cmd-injection
}
